src/sentry/conf/server.py

```diff
@@ -1921,7 +1921,11 @@ def build_cdc_postgres_init_db_volume(settings):
 STATUS_PAGE_ID = None
 STATUS_PAGE_API_HOST = "statuspage.io"
 
-SENTRY_ONPREMISE = True
+# Downstream apps (getsentry, single-tenant) set SENTRY_ONPREMISE, so keep
+# reading from that for now, but in the rest of _this_ codebase we should
+# reference SENTRY_SELF_HOSTED.
+SENTRY_ONPREMISE = True  # deprecated, use ...
+SENTRY_SELF_HOSTED = SENTRY_ONPREMISE  # ... this instead
```

Comment:
This is all wrong. I need to actually test this in `getsentry`. How do I do that?

The result of this PR will be to always set `SENTRY_SELF_HOSTED` to true, regardless of the value of `SENTRY_ONPREMISE` in a downstream project. Since `SENTRY_SELF_HOSTED` then becomes the config passed out the frontend, if I merge/deploy this then I will effectively turn SaaS into a self-hosted instance.

Reply:
@chadwhitacre we can add `SENTRY_SELF_HOSTED = False` here https://github.com/getsentry/getsentry/blob/40972e7b6f8c369da2324a42d01196040d53e2ca/getsentry/conf/settings/defaults.py#L29

Reply:
The better way to do this is to:

1. Set `SENTRY_SELF_HOSTED` in the downstream apps (`getsentry` and `single-tenant`) as a no-op duplicate. It sits right next to `SENTRY_ONPREMISE` but nothing reads it.
2. Modify the upstream (i.e., `sentry`, here) to use/read `SENTRY_SELF_HOSTED` on the backend (basically this here PR here).
3. Remove `SENTRY_ONPREMISE` from the downstreams.
4. Modify the upstream to set `isSelfHosted` as a no-op duplicate on the frontend.
5. Modify the downstreams to read from `isSelfHosted`.
6. Modify the upstream to read from `isSelfHosted` and remove `isOnPremise`.

The crux is that downstreams _set_ the backend `SENTRY_FOO` but _consume_ the frontend `isFooBar`, while upstreams consume the backend and set the frontend. The motion ends up looking like this:

```
backend  add    downstream
backend  add    upstream
backend  remove upstream
backend  remove downstream

frontend add    upstream
frontend add    downstream
frontend remove downstream
frontend remove upstream
```

Reply:
I've folded this back into https://github.com/getsentry/self-hosted/issues/796. To be continued over there ...